src/components/List.js

```diff
@@ -3,39 +3,150 @@ import PropTypes from 'prop-types';
 
 import { listItemTypes } from './utils';
 
-const List = ({ items,
-    activeItem,
-    ListItem,
-    idAttribute = "id",
-    className = '',
-    onClick,
-    itemsType = listItemTypes.OBJECT,
-    ...props }) => {
-    return (<div className={className}>
-        {items.map((item, idx) => {
-
-            let isActive = false;
-            if (itemsType === listItemTypes.PRIMITIVE) {
-                isActive = item === activeItem;
-            } else {
-                isActive = typeof activeItem === 'object' ?
-                    activeItem[idAttribute] === item[idAttribute] : activeItem === item[idAttribute];
-
-            }
-
-
-            const key = itemsType === listItemTypes.PRIMITIVE ? item : item[idAttribute];
-
-            return <ListItem
-                itemData={item}
-                idAttribute={key}
-                onClick={onClick ? onClick : null}
-                isActive={isActive}
-                idx={idx}
-                {...props}
-                key={key} />
-        })}
-    </div>)
+class List extends React.Component{
+    constructor(props) {
+        super(props);
+        this.state = { viewLess: false };     
+    }
+
+    toggleViewLess = () => {
+        this.setState((existingState) => {
+          let currentViewLess = existingState.viewLess;
+          return {
+            ...existingState,
+            viewLess: !currentViewLess
+          };
+        });
+      }
+
+    render (){
+
+        const {
+            items,
+            activeItem,
+            ListItem,
+            idAttribute = "id",
+            className = '',
+            onClick,
+            itemsType = listItemTypes.OBJECT,
+            facetName,
+            combinedFacets,
+            multiLevelField,
+            bucketedFacets,
+            displayName,
+            isRangeOpen,
+            textFacets
+        } = this.props
+
+        const { viewLess } = this.state
+
+        return (
+        <div>
+            <div className={`${className} ${viewLess? "UNX-facet__listShowLimited":""}` }>
+            {items.map((item, idx) => {
+
+                let isActive = false;
+                if (itemsType === listItemTypes.PRIMITIVE) {
+                    isActive = item === activeItem;
+                } else {
+                    isActive = typeof activeItem === 'object' ?
+                        activeItem[idAttribute] === item[idAttribute] : activeItem === item[idAttribute];
+
+                }
+
+
+                const key = itemsType === listItemTypes.PRIMITIVE ? item : item[idAttribute];
+
+                return <ListItem
+                    itemData={item}
+                    idAttribute={key}
+                    onClick={onClick ? onClick : null}
+                    isActive={isActive}
+                    idx={idx}
+                    {...this.props}
+                    key={key} />
+            })}
+                
```

Comment:
ListItem is rendering individually and is unaware of any proprietary facet logic.
Facets logic is added in conditional code below.